docs/migration.md

```diff
@@ -3,198 +3,93 @@ title: Migration
+    `https://example.com/metrics` or `http://exmaple.com/metrics` to be 
+    represented as `https://example.com/metrics:443` and 
+    `http://example.com/metrics:80` respectively, add them to your target URLs
+    - `agent`
+      Instead use the dedicated `--agent` cli flag.
+
+  Prometheus v3 will log a warning if you continue to pass these to 
+  `--enable-feature`.
 
 ## PromQL
 
-The following features have been removed from PromQL:
-
-- `drop_common_labels` function - the `without` aggregation modifier should be used
-  instead.
-- `keep_common` aggregation modifier - the `by` modifier should be used instead.
-- `count_scalar` function - use cases are better handled by `absent()` or correct
-  propagation of labels in operations.
-
-See [issue #3060](https://github.com/prometheus/prometheus/issues/3060) for more
-details.
+- The `.` pattern in regular expressions in PromQL matches newline characters. 
+  With this change a regular expressions like `.*` matches strings that include 
+  `\n`. This applies to matchers in queries and relabel configs. For example the 
+  following regular expressions now match the accompanying strings, wheras in 
+  Prometheus v2 these combinations didn't match.
+
+| Regex      | Additional matches  |
+| -----      | ------              |
+| ".*"       | "foo\n", "Foo\nBar" |
+| "foo.?bar" | "foo\nbar"          |
+| "foo.+bar" | "foo\nbar"          |
+
+  If you want Prometheus v3 to behave like v2 did, you will have to change your 
+  regular expressions by replacing all `.` patterns with `[^\n]`, e.g.  
+  `foo[^\n]*`.
+- Lookback and range selectors are left open and right closed (previously left 
+  closed and right closed). This change affects queries when the evaluation time 
+  perfectly aligns with the sample timestamps. For example assume querying a 
+  timeseries with even spaced samples exactly 1 minute apart. Before Prometheus 
+  3.x, range query with `5m` will mostly return 5 samples. But if the query 
+  evaluation aligns perfectly with a scrape, it would return 6 samples. In 
+  Prometheus 3.x queries like this will always return 5 samples.
+  This change has likely few effects for everyday use, except for some sub query 
+  use cases.
+  Query front-ends that align queries usually align sub-queries to multiples of 
+  the step size. These sub queries will likely be affected.
+  Tests are more likely to affected. To fix those either adjust the expected 
+  number of samples or extend to range by less then one sample interval.
+- The `holt_winters` function has been renamed to `double_exponential_smoothing` 
+  and is now guarded by the `promql-experimental-functions` feature flag.
+  If you want to keep using holt_winters, you have to do both of these things:
+    - Rename holt_winters to double_exponential_smoothing in your queries.
+    - Pass `--enable-feature=promql-experimental-functions` in your Prometheus 
+      cli invocation..
 
 ## Miscellaneous
 
```

Comment:
Tried to make a PR to this PR but lost in a world of cascading PRs. May just be easier to supply the suggested text here:

### Scrape protocols
Prometheus v3 is more strict concerning the `Content-Type` header received when
scraping. Prometheus v2 would default to the standard Prometheus text protocol
if the target being scraped did not specify a `Content-Type` header or if the
header was unparsable or unrecognised. This could lead to incorrect data being
parsed in the scrape. Prometheus v3 will now fail the scrape in such cases.

If a scrape target is not providing the correct `Content-Type` header the
fallback protocol can be specified using the `fallback_scrape_protocol`
parameter. See [Prometheus scrape_config documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).

This is a breaking change as scrapes that may have succeeded with Prometheus v2
may now fail if this fallback protocol is not specified.

Reply:
added